specification/search/resource-manager/Microsoft.Search/preview/2019-10-01-preview/search.json

```diff
@@ -777,6 +828,266 @@
+        },
+        "parameters": [
+          {
+            "$ref": "#/parameters/ClientRequestIdParameter"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/SearchServiceNameParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "The operation succeeded. The response contains the list of all private endpoint connections for the given service.",
+            "schema": {
+              "$ref": "#/definitions/PrivateEndpointConnectionListResult"
+            }
+          },
+          "default": {
+            "description": "An unexpected error occurred during the operation.",
+            "schema": {
+              "$ref": "#/definitions/CloudError"
+            }
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Search/searchServices/{searchServiceName}/sharedPrivateLinkResources/{sharedPrivateLinkResourceName}": {
+      "put": {
+        "tags": [
+          "SharedPrivateLinkResources"
+        ],
+        "operationId": "SharedPrivateLinkResources_Update",
+        "x-ms-examples": {
+          "SharedPrivateLinkResourceUpdate": {
+            "$ref": "./examples/UpdateSharedPrivateLinkResource.json"
+          }
+        },
+        "description": "Updates a shared private link resource managed by the Search service in the given resource group.",
+        "externalDocs": {
+          "url": "https://aka.ms/search-manage"
+        },
+        "parameters": [
+          {
+            "$ref": "#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/SearchServiceNameParameter"
+          },
+          {
+            "$ref": "#/parameters/SharedPrivateLinkResourceNameParameter"
+          },
+          {
+            "name": "sharedPrivateLinkResource",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/SharedPrivateLinkResource"
+            },
+            "description": "The definition of the shared private link resource to update."
+          },
+          {
+            "$ref": "#/parameters/ClientRequestIdParameter"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "The existing shared private link resource definition was successfully updated.",
+            "schema": {
+              "$ref": "#/definitions/SharedPrivateLinkResource"
+            }
+          },
+          "default": {
```

Comment:
Can you call out that we can also possibly return 409 Conflict under a different scenario.
This happens when someone tries to immediately create a shared private link resource with the same name after it has been deleted. This is because when we delete the shared private link resource ARM actually returns a 202, and we don't yet poll/provide a URI for customers to poll. We return immediately.

Usually, this only takes less than a minute -

Reply:
it is not required to list all the error scenarios.. just default error response is fine.

---
In reply to: [387213896](https://github.com/Azure/azure-rest-api-specs/pull/8572#discussion_r387213896) [](ancestors = 387213896)